code/modules/shuttle/mobile_instance.dm

```diff
@@ -0,0 +1,24 @@
+
+//ID is a temp ID that links shuttles in the map file.
+//Port and consoles must be on the same map file, will not work if the shuttle console is placed on a different map.
+//After map load, the shuttle will be given a new unique ID and any consoles with the old port will be given the new unique ID.
+//Used for making ships that can spawn multiple times and are fully contained in 1 file.
+//
+//Works for shuttles carrying other shuttles too.
+/obj/docking_port/mobile/instance/Initialize(mapload)
+	. = ..()
+	return INITIALIZE_HINT_LATELOAD
+
+/obj/docking_port/mobile/instance/LateInitialize()
+	var/static/number = 0
+	//Generate a unique ID
+	var/old_id = id
+	id = "[id]_[number++]"
+	//Find attached flight consoles
+	for(var/atom/thing as() in SSshuttle.consoles)
```

Comment:
IIRC LateInitialize can start firing before everything being loaded has Initialized, creating a race condition here.

But please double check if that's been fixed.

Reply:
I believe since its coming from a template spawn, initialize should be passed mapload instead of an atom which causes late initialize to be done after